src/pins/PinsStatuses.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react'
+import { withTranslation } from 'react-i18next'
+import GlyphCancel from '../icons/GlyphCancel'
+import GlyphPinCloud from '../icons/GlyphPinCloud'
+import '../files/file/PendingAnimation.css'
+
+const Status = {
+  Pending: 0,
+  Completed: 1,
+  Failed: 2
+}
+
+const Pin = (t, pin, status = Status.Pending, onDismiss) => {
+  const [service, cid] = pin.split(':')
+
+  let cloudClass = 'pa1'
+  let alt = ''
+  let dismissAlt = t('clickToDismiss')
+  if (status === Status.Pending) {
+    cloudClass += ' fill-aqua PendingAnimation'
+    alt = t('pinningRemotely')
+    dismissAlt = t('clickToCancel')
+  } else if (status === Status.Completed) {
+    cloudClass += ' fill-aqua'
+    alt = t('pinningCompleted')
+  } else if (status === Status.Failed) {
+    cloudClass += ' fill-red'
+    alt = t('pinningFailed')
+  }
+
+  return (<li className="flex w-100 bb b--light-gray items-center f6 charcoal" key={cid}>
+    <GlyphPinCloud className={cloudClass} style={{ width: '36px' }} alt={alt} title={alt} />
+    <span className="truncate">{cid}</span>
+    <span className='gray mh2'>| {service}</span>
+    <span className='br-100 o-70 ml-auto w2 h2' alt={dismissAlt} title={dismissAlt} >
+      <button onClick={() => onDismiss(pin)} className='w2 h2 pa0'>
+        <GlyphCancel className='fill-aqua' />
+      </button>
+    </span>
+  </li>)
+}
+
+const PinsStatuses = ({ pendingPins, failedPins, completedPins, onDismissFailedPin, onDismissCompletedPin, doCancelPendingPin, t }) => {
+  return (
+    <div>
+      <h3>Pins</h3>
```

Comment:
This should be added to locale files and have two values:
- "All done, no pins in progress" (when list is empty)
- "Pinning in progress"

Reply:
@lidel you mean, add local pins to this page, right?

Reply:
I meant translation files https://github.com/ipfs/ipfs-webui/tree/main/public/locales/en :)

Reply:
Oeh, done 😃